compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -154,4 +154,6 @@ pub fn provide(providers: &mut Providers) {
     };
     providers.in_scope_traits_map =
         |tcx, id| tcx.hir_crate(()).owners[id].as_owner().map(|owner_info| &owner_info.trait_map);
+    providers.impl_parent =
+        |tcx, def_id| tcx.hir().impl_parent(def_id.expect_local()).map(|def_id| def_id.to_def_id());
```

Comment:
IIUC, `impl_parent` is about specialization, not about finding which impl block an item corresponds to.